integrations/cohere/src/cohere_haystack/embedders/document_embedder.py

```diff
@@ -0,0 +1,149 @@
+                             to keep the logs clean.
+        :param metadata_fields_to_embed: List of meta fields that should be embedded along with the Document text.
+        :param embedding_separator: Separator used to concatenate the meta fields to the Document text.
+        """
+
+        if api_key is None:
+            try:
+                api_key = os.environ["COHERE_API_KEY"]
+            except KeyError as error_msg:
+                msg = (
+                    "CohereDocumentEmbedder expects an Cohere API key. Please provide one by setting the environment "
+                    "variable COHERE_API_KEY (recommended) or by passing it explicitly."
+                )
+                raise ValueError(msg) from error_msg
+
+        self.api_key = api_key
+        self.model_name = model_name
+        self.api_base_url = api_base_url
+        self.truncate = truncate
+        self.use_async_client = use_async_client
+        self.max_retries = max_retries
+        self.timeout = timeout
+        self.batch_size = batch_size
+        self.progress_bar = progress_bar
+        self.metadata_fields_to_embed = metadata_fields_to_embed or []
+        self.embedding_separator = embedding_separator
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary omitting the api_key field.
+        """
+        return default_to_dict(
+            self,
+            model_name=self.model_name,
+            api_base_url=self.api_base_url,
+            truncate=self.truncate,
+            use_async_client=self.use_async_client,
+            max_retries=self.max_retries,
+            timeout=self.timeout,
+            batch_size=self.batch_size,
+            progress_bar=self.progress_bar,
+            metadata_fields_to_embed=self.metadata_fields_to_embed,
+            embedding_separator=self.embedding_separator,
+        )
+
+    def _prepare_texts_to_embed(self, documents: List[Document]) -> List[str]:
+        """
+        Prepare the texts to embed by concatenating the Document text with the metadata fields to embed.
+        """
+        texts_to_embed: List[str] = []
+        for doc in documents:
+            meta_values_to_embed = [
+                str(doc.meta[key]) for key in self.metadata_fields_to_embed if doc.meta.get(key) is not None
+            ]
+
+            text_to_embed = self.embedding_separator.join(meta_values_to_embed + [doc.content or ""])  # noqa: RUF005
+            texts_to_embed.append(text_to_embed)
+        return texts_to_embed
+
+    @component.output_types(documents=List[Document], metadata=Dict[str, Any])
+    def run(self, documents: List[Document]):
+        """
+        Embed a list of Documents.
+        The embedding of each Document is stored in the `embedding` field of the Document.
+
+        :param documents: A list of Documents to embed.
+        """
+
+        if not isinstance(documents, list) or not isinstance(documents[0], Document):
```

Comment:
Let's change this to 
`if not isinstance(documents, list) or documents and not isinstance(documents[0], Document):`so that empty lists do not raise an error. If an empty list is provided as input, components should also return an empty list as output.

Reply:
Ideally, we'd also have a test case to check that behavior.